### PR TITLE
fix #215

### LINKED ISSUE #215: compilation error on PHP 8.1beta2
Not sure if 8.1 will be supported but compiling it currently fails with the following error:
```/tmp/parallel/src/runtime.c:161:42: error: 'zend_class_serialize_deny' undeclared (first use in this function); did you mean 'zend_ce_serializable'?```

### PR DESCRIPTION
**Pull request by WyriHaximus**
Resolves #215

### CODE REVIEW

File: src/channel.c [hunk 1]
```diff
@@ -382,9 +382,14 @@ PHP_MINIT_FUNCTION(PARALLEL_CHANNEL)
     php_parallel_channel_ce = zend_register_internal_class(&ce);
     php_parallel_channel_ce->create_object = php_parallel_channel_create;
     php_parallel_channel_ce->ce_flags |= ZEND_ACC_FINAL;
+#if PHP_VERSION_ID >= 80100
+    php_parallel_channel_ce->ce_flags |= ZEND_ACC_NOT_SERIALIZABLE ;
```
Comment (cmb69) [inline src/channel.c]: nit: remove trailing space:
```suggestion
    php_parallel_channel_ce->ce_flags |= ZEND_ACC_NOT_SERIALIZABLE ;
```
```suggestion
    php_parallel_channel_ce->ce_flags |= ZEND_ACC_NOT_SERIALIZABLE;
```

File: src/channel.c [hunk 2]
```diff
@@ -382,9 +382,14 @@ PHP_MINIT_FUNCTION(PARALLEL_CHANNEL)
...
+#endif
 
+#if PHP_VERSION_ID < 80100
```
Comment (cmb69) [inline src/channel.c]: Can combine these directives:
```suggestion
#elif
```

**Comment by WyriHaximus** [comment]
@cmb69 Thanks, applied your suggestions

File: src/channel.c (hunk 2 above)
Comment (WyriHaximus) [reply src/channel.c]: Had to revert this one because CIi failed

Comment (cmb69) [reply src/channel.c]: Oh, sorry, that should have been `#else`.

Comment (WyriHaximus) [reply src/channel.c]: Doh, should have caught that 🤦

File: src/events.c [hunk 3]
```diff
@@ -269,9 +269,14 @@ PHP_MINIT_FUNCTION(PARALLEL_EVENTS)
     php_parallel_events_ce->create_object = php_parallel_events_create;
     php_parallel_events_ce->get_iterator  = php_parallel_events_loop_create;
     php_parallel_events_ce->ce_flags |= ZEND_ACC_FINAL;
+#if PHP_VERSION_ID >= 80100
+    php_parallel_events_ce->ce_flags |= ZEND_ACC_NOT_SERIALIZABLE ;
+#endif
 
+#if PHP_VERSION_ID < 80100
```
Comment (cmb69) [inline src/events.c:276]:
```suggestion
#else
```

File: src/future.c [hunk 4]
```diff
@@ -298,9 +298,14 @@ PHP_MINIT_FUNCTION(PARALLEL_FUTURE)
     php_parallel_future_ce = zend_register_internal_class(&ce);
     php_parallel_future_ce->create_object = php_parallel_future_create;
     php_parallel_future_ce->ce_flags |= ZEND_ACC_FINAL;
+#if PHP_VERSION_ID >= 80100
+    php_parallel_future_ce->ce_flags |= ZEND_ACC_NOT_SERIALIZABLE ;
+#endif
 
+#if PHP_VERSION_ID < 80100
```
Comment (cmb69) [inline src/future.c:305]:
```suggestion
#else
```

File: src/input.c [hunk 5]
```diff
@@ -194,9 +194,14 @@ PHP_MINIT_FUNCTION(PARALLEL_EVENTS_INPUT)
     php_parallel_events_input_ce = zend_register_internal_class(&ce);
     php_parallel_events_input_ce->create_object = php_parallel_events_input_create;
     php_parallel_events_input_ce->ce_flags |= ZEND_ACC_FINAL;
+#if PHP_VERSION_ID >= 80100
+    php_parallel_events_input_ce->ce_flags |= ZEND_ACC_NOT_SERIALIZABLE ;
+#endif
 
+#if PHP_VERSION_ID < 80100
```
Comment (cmb69) [inline src/input.c:201]:
```suggestion
#else
```

File: src/runtime.c [hunk 6]
```diff
@@ -157,9 +157,14 @@ PHP_MINIT_FUNCTION(PARALLEL_RUNTIME)
     php_parallel_runtime_ce = zend_register_internal_class(&ce);
     php_parallel_runtime_ce->create_object = php_parallel_runtime_create;
     php_parallel_runtime_ce->ce_flags |= ZEND_ACC_FINAL;
+#if PHP_VERSION_ID >= 80100
+    php_parallel_runtime_ce->ce_flags |= ZEND_ACC_NOT_SERIALIZABLE ;
+#endif
 
+#if PHP_VERSION_ID < 80100
```
Comment (cmb69) [inline src/runtime.c:164]:
```suggestion
#else
```

File: src/sync.c [hunk 7]
```diff
@@ -311,9 +311,14 @@ PHP_MINIT_FUNCTION(PARALLEL_SYNC)
     php_parallel_sync_ce = zend_register_internal_class(&ce);
     php_parallel_sync_ce->create_object = php_parallel_sync_object_create;
     php_parallel_sync_ce->ce_flags |= ZEND_ACC_FINAL;
+#if PHP_VERSION_ID >= 80100
+    php_parallel_sync_ce->ce_flags |= ZEND_ACC_NOT_SERIALIZABLE ;
+#endif
 
+#if PHP_VERSION_ID < 80100
```
Comment (cmb69) [inline src/sync.c:318]:
```suggestion
#else
```

**Review by cmb69 — CHANGES_REQUESTED** [review]
We should use `#else` elsewhere, too. :)

**Comment by dixyes** [comment]
should we use `#ifdef ZEND_ACC_NOT_SERIALIZABLE` ? like [this](https://github.com/dixyes/parallel/commit/396925e617ec0a8e3a6240575ff8efc7cdd20a3c)

**Comment by cmb69** [comment]
Oops, totally forgot about this PR, and just merged #229, which solves the same issue. Thanks for your work on this nonetheless!

**Comment by WyriHaximus** [comment]
@cmb69 No problem, does this mean you've taken over maintenance of this extension?

**Comment by cmb69** [comment]
No, not really; I'm just helping a bit where I can. :)

**Comment by WyriHaximus** [comment]
Still great, this project could use some love and run on PHP 8.x

**Pull request closed without merging**